pkg/opensearch/stream_handler.go

```diff
@@ -0,0 +1,170 @@
+			Status: backend.SubscribeStreamStatusNotFound,
+		}, fmt.Errorf("missing refId in stream path: %s", req.Path)
+	}
+
+	_, err := o.getDSInfo(ctx, req.PluginContext)
+	if err != nil {
+		o.logger.Error("SubscribeStream: failed to get datasource info", "error", err)
+		return &backend.SubscribeStreamResponse{
+			Status: backend.SubscribeStreamStatusNotFound,
+		}, fmt.Errorf("failed to get datasource info: %w", err)
+	}
+
+	o.logger.Info("SubscribeStream: path validated, authorizing stream", "path", req.Path)
+	return &backend.SubscribeStreamResponse{
+		Status: backend.SubscribeStreamStatusOK,
+	}, nil
+}
+
+func (o *OpenSearchDatasource) RunStream(ctx context.Context, req *backend.RunStreamRequest, sender *backend.StreamSender) error {
+	o.logger.Info("RunStream (Polling Mode) called", "path", req.Path)
+	o.logger.Debug("RunStream", "full_req_details", req)
+
+	if !strings.HasPrefix(req.Path, "tail/") {
+		o.logger.Error("RunStream: invalid path format", "path", req.Path)
+		return fmt.Errorf("RunStream: invalid path format for stream: %s, expected 'tail/{refId}'", req.Path)
+	}
+	refId := strings.TrimPrefix(req.Path, "tail/")
+	if refId == "" {
+		o.logger.Error("RunStream: missing refId in path after trim", "path", req.Path)
+		return fmt.Errorf("RunStream: missing refId in stream path after trim: %s", req.Path)
+	}
+
+	// Get query data directly from the stream request (standard Grafana pattern)
+	rawQueryJSON := req.Data
+	if len(rawQueryJSON) == 0 {
+		o.logger.Error("RunStream: no query data provided in stream request", "refId", refId, "path", req.Path)
+		return fmt.Errorf("no query data provided for streaming refId: %s", refId)
+	}
+
+	o.logger.Info("RunStream: starting polling for query", "refId", refId, "rawQuery", string(rawQueryJSON))
+
+	ticker := time.NewTicker(2 * time.Second) // Faster polling for more responsive live tailing
+	defer ticker.Stop()
+
+	lastTo := time.Now().Add(-1 * time.Second) // Start slightly in the past to catch any timing issues
+
+	for {
+		select {
+		case <-ctx.Done():
+			o.logger.Info("RunStream: context canceled, stopping stream", "refId", refId)
+			return ctx.Err()
+		case <-ticker.C:
+			currentTime := time.Now()
+			backendQuery := backend.DataQuery{
+				RefID:     refId,
+				TimeRange: backend.TimeRange{From: lastTo, To: currentTime},
+				JSON:      rawQueryJSON,
+			}
+
+			if !backendQuery.TimeRange.To.After(backendQuery.TimeRange.From) {
+				o.logger.Debug("RunStream: 'To' time is not after 'From' time, skipping poll to avoid empty range", "from", backendQuery.TimeRange.From, "to", backendQuery.TimeRange.To)
+				continue
+			}
+
+			o.logger.Info("RunStream: Polling OpenSearch", "refId", refId, "from", backendQuery.TimeRange.From, "to", backendQuery.TimeRange.To, "duration", backendQuery.TimeRange.To.Sub(backendQuery.TimeRange.From))
+
+			osClient, err := client.NewClient(ctx, req.PluginContext.DataSourceInstanceSettings, o.httpClient, &backendQuery.TimeRange)
```

Comment:
I think we we should be able to just run 
```
queryDataResponse, err :=  o.QueryData(ctx, &backend.QueryDataRequest{
				Queries:  []backend.DataQuery{backendQuery},
				PluginContext: req.PluginContext,	
				Headers:  req.Headers,
			})
```
instead of this and the following lines, since we have all the data to construct the data query request.